pkg/kmm/preflightvalidationocp_test.go

```diff
@@ -0,0 +1,416 @@
+
+		if testCase.expectedErr == "" {
+			assert.NotNil(t, testBuilder)
+			assert.Equal(t, testCase.expectedPreflight.Name, testBuilder.Definition.Name)
+			assert.Equal(t, testCase.expectedPreflight.Namespace, testBuilder.Definition.Namespace)
+		} else {
+			assert.Equal(t, testCase.expectedErr, testBuilder.errorMsg)
+		}
+	}
+}
+
+func TestPreflightValidationWithReleaseImage(t *testing.T) {
+	testCases := []struct {
+		image       string
+		expecterErr string
+	}{
+		{
+			image:       "some-image",
+			expecterErr: "",
+		},
+		{
+			image:       "",
+			expecterErr: "invald 'image' argument can not be nil",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := buildValidTestPreflight(buildPreflightTestClientWithDummyObject())
+		testBuilder.WithReleaseImage(testCase.image)
+
+		if testCase.expecterErr == "" {
+			assert.Equal(t, testBuilder.Definition.Spec.ReleaseImage, testCase.image)
+		} else {
+			assert.Equal(t, testCase.expecterErr, testBuilder.errorMsg)
+		}
+	}
+}
+
+func TestPreflightWithUseRTKernel(t *testing.T) {
+	testCases := []struct {
+		useRT bool
+	}{
+		{
+			useRT: true,
+		},
+		{
+			useRT: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := buildValidTestPreflight(buildPreflightTestClientWithDummyObject())
+		testBuilder.WithUseRTKernel(testCase.useRT)
+		assert.Equal(t, testCase.useRT, testBuilder.Definition.Spec.UseRTKernel)
+	}
+}
+
+func TestPreflightWithPushBuiltImage(t *testing.T) {
+	testCases := []struct {
+		push bool
+	}{
+		{
+			push: true,
+		},
+		{
+			push: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := buildValidTestPreflight(buildPreflightTestClientWithDummyObject())
+		testBuilder.WithPushBuiltImage(testCase.push)
+		assert.Equal(t, testCase.push, testBuilder.Definition.Spec.PushBuiltImage)
+	}
+}
+
+func TestPreflightWithOptions(t *testing.T) {
+	testBuilder := buildValidTestPreflight(buildPreflightTestClientWithDummyObject())
+
+	testBuilder.WithOptions(func(builder *PreflightValidationOCPBuilder) (*PreflightValidationOCPBuilder, error) {
+		return builder, nil
+	})
+
+	assert.Equal(t, "", testBuilder.errorMsg)
+
+	testBuilder.WithOptions(func(builder *PreflightValidationOCPBuilder) (*PreflightValidationOCPBuilder, error) {
+		return builder, fmt.Errorf("error")
+	})
+
+	assert.Equal(t, "error", testBuilder.errorMsg)
+}
+
+func TestPullPreflightValidationOCP(t *testing.T) {
+	testCases := []struct {
+		name                string
+		namespace           string
+		expectedError       bool
+		addToRuntimeObjects bool
+		expectedErrorText   string
+		client              bool
+	}{
+		{
+			name:                "test",
+			namespace:           "testns",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "preflightvalidation 'apiClient' cannot be empty",
+			client:              false,
+		},
+		{
+			name:                "test",
```

Comment:
Something is wrong here.
This is exactly the same as last test case but in last test case you have error.

Reply:
Indeed. The `expectedError` and `expectedErrorText` were also misleading. 

Fixed it. Thanks !